### DISCUSSION TITLE
What's the best endpoint to get list of cases, filtered by district and nature of suit, along with the complaints for those cases?

### DISCUSSION THREAD

**Discussion by ghing** [freelawproject/courtlistener#3239] [Q&A] [answered]
I have a few questions about the API that I'm hoping someone could answer.

- What endpoint should I use if I want to retrieve all cases in a particular district that match a particular nature of suit pattern? I'm looking for results similar to this search using the web form: https://www.courtlistener.com/?q=&type=r&order_by=dateFiled%20desc&nature_of_suit=Prisoner%3APrison%20conditions&court=azd
- Is it possible to limit the results to an office within the district (e.g. just the Tucson office and not Phoenix)? I see that this information is available on the IDB tab of a case's information, but I'm not sure if it's filterable by other endpoints.
- If it's possible to retrieve a list of cases, is it also possible to retrieve a URL to their complaints, either in the RECAP archive, if available, or in PACER?

**Comment by mlissner** [comment, marked as answer]
Thanks for asking. A few replies:

>     * What endpoint should I use if I want to retrieve all cases in a particular district that match a particular nature of suit pattern? I'm looking for results similar to this search using the web form: 

Probably the easiest thing to do with that is convert it to an API request by changing the URL a bit. This has the same GET params but returns JSON instead of HTML:

https://www.courtlistener.com/api/rest/v3/search/?q=&type=r&order_by=dateFiled%20desc&nature_of_suit=Prisoner%3APrison%20conditions&court=azd

You could also tweak this to get back one result per case by changing it to `type=d` (that only works on API results, I believe).

>     * Is it possible to limit the results to an office within the district (e.g. just the Tucson office and not Phoenix)? I see that this information is available on the IDB tab of a case's information, but I'm not sure if it's filterable by other endpoints.

I'm sorry, I don't think that's really possible now. We don't make the FJC data searchable or filterable as far as I'm aware.

>     * If it's possible to retrieve a list of cases, is it also possible to retrieve a URL to their complaints, either in the RECAP archive, if available, or in PACER?

Well....if you have a list of cases, there's no surefire way to get their initial complaints, but they're usually document 1. So if you assume that, what I'd do is an API request for each of those documents. 

So if you have docket number `4214664`, you could do something like this:

https://www.courtlistener.com/api/rest/v3/recap-documents/?docket_entry__docket=4214664&document_number=1&document_type=1

That gets you:

 - `docket_entry__docket`: The correct docket
 - `document_number`: Docket entry number 1
 - `document_type`: The document, not a PACER attachment

In the response from this, you'll see:

```
            "filepath_local": "recap/gov.uscourts.dcd.178502/gov.uscourts.dcd.178502.1.0_48.pdf",
```

Tack that onto storage.courtlistener.com, and you've got the download URL:

https://storage.courtlistener.com/recap/gov.uscourts.dcd.178502/gov.uscourts.dcd.178502.1.0_48.pdf

Great! 

---

But if you don't get back the filepath_local value, you'll want to buy the document. We don't provide a link to PACER for that, but we do have an API that you can use to buy it instead of doing it by hand. Search here for "Fetch" and you should see some good examples of using our Fetch API (it's documented too).

Does that help
